notebooks/MERFISH_Pipeline_-_U2O2_Cell_Culture_-_1_FOV.py

```diff
@@ -110,24 +120,31 @@
 # EPY: END code
 
 # EPY: START markdown
-# Use MERFISH-calculated size factors to scale the channels across the hybridization rounds and visualize the resulting filtered and scaled images
+# Use MERFISH-calculated size factors to scale the channels across the hybridization rounds and visualize the resulting filtered and scaled images. Right now we have to extract this information from the metadata and apply this transformation manually.
 # EPY: END markdown
 
 # EPY: START code
-stack_blurred = s.image.numpy_array
-sc = s.org['metadata']['scale']
-sc_df = pd.DataFrame([(int(k), v) for k,v in sc.items()], columns = ['bit', 'scale'])
-mp = pd.merge(s.squeeze_map, sc_df, on='bit', how='left')
-scale_dict = dict(zip(mp.ind.values, mp.scale.values))
+scale_factors = {(t[Indices.HYB], t[Indices.CH]): t['scale_factor'] for index, t in s.tile_metadata.iterrows()}
+# EPY: END code
 
-for k, v in scale_dict.items():
-    stack_blurred[k] = stack_blurred[k]/v
-    
-s.set_stack(s.un_squeeze(stack_blurred))
+# EPY: START code
+# this is a scaling method. It would be great to use image.apply here. It's possible, but we need to expose H & C to 
+# at least we can do it with get_slice and set_slice right now.
+
+for indices in s.image._iter_indices():
+    data = s.image.get_slice(indices)[0]
+    scaled = data / scale_factors[indices[Indices.HYB], indices[Indices.CH]]
+    s.image.set_slice(indices, scaled)
+# EPY: END code
+
+# EPY: START code
+from scipy.stats import scoreatpercentile
```

Comment:
this is an odd snippet

Reply:
Could you explain what part you find odd?

Reply:
it's a cell with one line, and it's an import.  unless i missed something.

Reply:
Got it, I'll move it to the imports section. I was wondering if you were talking about the code I added.